### PR TITLE
Improve speed of Intel AVX1/2 ASM for SHA-256 and SHA-512

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

**Comment by toddouska** [comment]
Can you resolve the conflicts please?

File: wolfssl/wolfcrypt/sha256.h [hunk 1]
```diff
@@ -96,6 +104,7 @@ typedef struct wc_Sha256 {
     word32  loLen;     /* length in bytes   */
     word32  hiLen;     /* length in bytes   */
     void*   heap;
+    const byte* data;
```
Comment (toddouska) [inline wolfssl/wolfcrypt/sha256.h]: Is this pointer needed in non intel asm cases?  I don't like to add size to these structures unless we have to.

File: wolfssl/wolfcrypt/sha512.h [hunk 2]
```diff
@@ -78,12 +86,13 @@ enum {
...
 typedef struct wc_Sha512 {
+    word64  digest[WC_SHA512_DIGEST_SIZE / sizeof(word64)];
+    word64  buffer[WC_SHA512_BLOCK_SIZE  / sizeof(word64)];
     word32  buffLen;   /* in bytes          */
     word64  loLen;     /* length in bytes   */
     word64  hiLen;     /* length in bytes   */
-    word64  digest[WC_SHA512_DIGEST_SIZE / sizeof(word64)];
-    word64  buffer[WC_SHA512_BLOCK_SIZE  / sizeof(word64)];
     void*   heap;
+    const byte* data;
```
Comment (toddouska) [inline wolfssl/wolfcrypt/sha512.h]: Is this pointer needed in non intel asm cases?  I don't like to add size to these structures unless we have to.

**Comment by toddouska** [comment]
This doesn't build for me on Mac to test speedups:

wolfcrypt/src/sha512.c:316:38: error: use of undeclared identifier
      'Transform_Sha512_AVX1_RORX'; did you mean 'Transform_Sha512_AVX1_Len'?
                Transform_Sha512_p = Transform_Sha512_AVX1_RORX;
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~
                                     Transform_Sha512_AVX1_Len
wolfcrypt/src/sha512.c:266:20: note: 'Transform_Sha512_AVX1_Len' declared here
        static int Transform_Sha512_AVX1_Len(wc_Sha512 *sha512, word32 len);
                   ^
wolfcrypt/src/sha512.c:316:36: error: incompatible pointer types assigning to
      'int (*)(wc_Sha512 *)' (aka 'int (*)(struct wc_Sha512 *)') from 'int
      (wc_Sha512 *, word32)' (aka 'int (struct wc_Sha512 *, unsigned int)')
      [-Werror,-Wincompatible-pointer-types]
                Transform_Sha512_p = Transform_Sha512_AVX1_RORX;
                                   ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~
wolfcrypt/src/sha512.c:317:42: error: use of undeclared identifier
      'Transform_Sha512_AVX1_RORX_Len'; did you mean
      'Transform_Sha512_AVX1_Len'?
                Transform_Sha512_Len_p = Transform_Sha512_AVX1_RORX_Len;
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                         Transform_Sha512_AVX1_Len
wolfcrypt/src/sha512.c:266:20: note: 'Transform_Sha512_AVX1_Len' declared here
        static int Transform_Sha512_AVX1_Len(wc_Sha512 *sha512, word32 len);
                   ^
  CC       wolfcrypt/src/src_libwolfssl_la-wc_port.lo
  CC       wolfcrypt/src/src_libwolfssl_la-error.lo

File: wolfssl/wolfcrypt/sha256.h (hunk 1 above)
Comment (SparkiDev) [reply wolfssl/wolfcrypt/sha256.h]: Removed

File: wolfssl/wolfcrypt/sha512.h (hunk 2 above)
Comment (SparkiDev) [reply wolfssl/wolfcrypt/sha512.h]: Removed

**Pull request merged**